plugins/source/aws/client/testing.go

```diff
@@ -24,77 +19,39 @@ type TestOptions struct {
 }
 
 func AwsMockTestHelper(t *testing.T, table *schema.Table, builder func(*testing.T, *gomock.Controller) Services, testOpts TestOptions) {
-	version := "vDev"
-
 	table.IgnoreInTests = false
 	t.Helper()
 	ctrl := gomock.NewController(t)
 	l := zerolog.New(zerolog.NewTestWriter(t)).Output(
 		zerolog.ConsoleWriter{Out: os.Stderr, TimeFormat: time.StampMicro},
 	).Level(zerolog.DebugLevel).With().Timestamp().Logger()
 
-	newTestExecutionClient := func(ctx context.Context, logger zerolog.Logger, spec specs.Source, _ source.Options) (schema.ClientMeta, error) {
-		var awsSpec Spec
-		if err := spec.UnmarshalSpec(&awsSpec); err != nil {
-			return nil, fmt.Errorf("failed to unmarshal aws spec: %w", err)
-		}
-		awsSpec.SetDefaults()
-		awsSpec.UsePaidAPIs = true
-		awsSpec.TableOptions = &testOpts.TableOptions
-		c := NewAwsClient(l, nil, &awsSpec)
-		services := builder(t, ctrl)
-		services.Regions = []string{"us-east-1"}
-		c.ServicesManager.InitServicesForPartitionAccount("aws", "testAccount", services)
-		c.Partition = "aws"
-		return &c, nil
+	var awsSpec Spec
+	awsSpec.SetDefaults()
+	awsSpec.UsePaidAPIs = true
+	awsSpec.TableOptions = &testOpts.TableOptions
+	c := NewAwsClient(l, &awsSpec)
+	services := builder(t, ctrl)
+	services.Regions = []string{"us-east-1"}
+	c.ServicesManager.InitServicesForPartitionAccount("aws", "testAccount", services)
+	c.Partition = "aws"
+	tables := schema.Tables{table}
+
+	if err := transformers.TransformTables(tables); err != nil {
+		t.Fatal(err)
 	}
 
-	p := source.NewPlugin(
-		table.Name,
-		version,
-		[]*schema.Table{
-			table,
-		},
-		newTestExecutionClient)
-	p.SetLogger(l)
-	source.TestPluginSync(t, p, specs.Source{
-		Name:         "dev",
-		Path:         "cloudquery/dev",
-		Version:      version,
-		Tables:       []string{table.Name},
-		Destinations: []string{"mock-destination"},
-	}, source.WithTestPluginAdditionalValidators(validateTagStructure))
```

Comment:
I think we lost the additional validator for tag structure in the transition here, and I'd say it's quite important we keep it in one form or another